server/statistics/region_collection.go

```diff
@@ -296,17 +280,35 @@ func getRegionLabelIsolation(stores []*core.StoreInfo, labels []string) string {
 }
 
 func notIsolatedStoresWithLabel(stores []*core.StoreInfo, label string) [][]*core.StoreInfo {
-	m := make(map[string][]*core.StoreInfo)
+	var emptyValueStores []*core.StoreInfo
+	valueStoresMap := make(map[string][]*core.StoreInfo)
+
 	for _, s := range stores {
 		labelValue := s.GetLabelValue(label)
 		if labelValue == "" {
-			continue
+			emptyValueStores = append(emptyValueStores, s)
+		} else {
+			valueStoresMap[labelValue] = append(valueStoresMap[labelValue], s)
+		}
+	}
+
+	if len(valueStoresMap) == 0 {
+		if len(emptyValueStores) > 1 {
+			return [][]*core.StoreInfo{emptyValueStores}
 		}
-		m[labelValue] = append(m[labelValue], s)
+		return nil
 	}
+
 	var res [][]*core.StoreInfo
-	for _, stores := range m {
-		if len(stores) > 1 {
+	if len(emptyValueStores) == 0 {
+		for _, stores := range valueStoresMap {
+			if len(stores) > 1 {
+				res = append(res, stores)
+			}
+		}
+	} else {
+		for _, stores := range valueStoresMap {
+			stores = append(stores, emptyValueStores...)
 			res = append(res, stores)
 		}
 	}
```

Comment:
What about adding some comments to these different situations? For example, for `valueStoresMap` is empty, it means no store has this label configured, for `emptyValueStores` is empty, it means every store has this label configured, and for `emptyValueStores` is not empty, it means this is at least one store has this label configured but not all.

Reply:
good idea